lib/datadog/remote-config/src/lib.rs

```diff
@@ -0,0 +1,481 @@
+    interchange,
+    metadata::{Metadata, MetadataPath, MetadataVersion, RawSignedMetadata},
+    repository::{EphemeralRepository, RepositoryProvider, RepositoryStorage},
+};
+
+mod proto {
+    include!(concat!(env!("OUT_DIR"), "/datadog.config.rs"));
+}
+mod error;
+
+use error::Result;
+use proto::{DelegatedMeta, LatestConfigsRequest, LatestConfigsResponse, TopMeta};
+
+pub use error::Error;
+pub use tuf::metadata::{TargetDescription, TargetPath};
+
+type TUFClient = tuf::client::Client<
+    interchange::Json,
+    EphemeralRepository<interchange::Json>,
+    EphemeralRepository<interchange::Json>,
+>;
+
+// TODO: are these safe to commit/embed?
+const CONFIG_ROOT: &[u8] = include_bytes!("../config_root.json");
+const DIRECTOR_ROOT: &[u8] = include_bytes!("../director_root.json");
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Config {
+    pub site: String,
+    pub api_key: String,
+    pub app_key: String,
+    pub hostname: String,
+    pub agent_version: String,
+}
+
+#[derive(Debug)]
+struct Inner {
+    http: hyper::Client<HttpsConnector<HttpConnector>>,
+    site: String,
+    api_key: String,
+    app_key: String,
+}
+
+pub struct Client {
+    inner: Inner,
+    hostname: String,
+    agent_version: String,
+    products: HashSet<String>,
+    new_products: HashSet<String>,
+    active_client: proto::Client,
+    backend_client_state: Vec<u8>,
+    config_client: TUFClient,
+    director_client: TUFClient,
+}
+
+impl Client {
+    pub async fn initialize(config: Config) -> Result<Self> {
+        let Config {
+            site,
+            api_key,
+            app_key,
+            hostname,
+            agent_version,
+        } = config;
+
+        let conn = HttpsConnector::new().map_err(|e| Error::OpenSsl(e.to_string()))?;
+        let inner = Inner {
+            http: hyper::Client::builder().build(conn),
+            site,
+            api_key,
+            app_key,
+        };
+
+        let request = LatestConfigsRequest {
+            hostname: hostname.clone(),
+            agent_version: agent_version.clone(),
+            ..Default::default()
+        };
+
+        // Send an initial request to fill out the rest of the metadata
+        let response = inner.send_request(request).await?;
+
+        // The correct meta versions here will get filled in on `update`
+        let client_state = proto::ClientState::default();
+
+        let client_agent = proto::ClientAgent {
+            name: String::from("vector"),
+            version: agent_version.clone(),
+        };
+
+        let id = Alphanumeric.sample_string(&mut rand::thread_rng(), 21);
+
+        let active_client = proto::Client {
+            state: Some(client_state),
+            id,
+            is_agent: false,
+            client_agent: Some(client_agent),
+            last_seen: std::time::SystemTime::now()
+                .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                .expect("unix epoch is in the past")
```

Comment:
Nit? As that's what we _expect_.
```suggestion
                .expect("unix epoch not to be in the past")
```